specs for offprings improved

katafira opened the pull request.

sergio-ocon commented on spec/models/offspring_spec.rb:
```diff
@@ -28,25 +28,26 @@
+    child=FactoryGirl.build(:offspring, :grade => :primary_second)
+    expect(child).to be_valid
 
-  it "has an grade that is an integer" do
```
has a grade that is an integer should be modified to only allow variables in the enum

alexvkcr replied:
"has a grade that is an integer" is already erased.

sergio-ocon commented on spec/models/offspring_spec.rb:
```diff
@@ -28,25 +28,26 @@
-    expect(child).not_to be_valid
-    expect(child.errors[:grade]).to include("tiene que ser entero")
+    child=FactoryGirl.build(:offspring, :grade => :primary_third)
+    expect(child).to be_valid
```
You can test all grades with a :grade.each do |x|
(on the enum values)
end

alexvkcr replied:
Thanks for this idea, we did two tests for model. One for 'primary_first' and another one for the rest of grades. We wanted to stress that all grades are accepted in the model but 'primary_first' has a different status (being the only accepted later in the controller).

We will change this to a single test proving every grade is accepted for the model.

sergio-ocon commented on spec/models/offspring_spec.rb:
```diff
@@ -28,25 +28,26 @@
+    child=FactoryGirl.build(:offspring, :grade => :primary_third)
+    expect(child).to be_valid
 
+    child=FactoryGirl.build(:offspring, :grade => :others)
```
Only offspring from first grade should be allowed to sign up:
However, I believe the model should not show this restriction (because it does not make sense to put it in the model), but it should be on the controller

alexvkcr replied:
We understand the model accepts children from differents grades but the operation in the controller only allows adding children from first grade, thus, we did tests to accept all-grade children in the model and accepting only 'primary_first' children in the controller.

sergio-ocon commented on spec/controllers/offsprings_controller_spec.rb:
```diff
@@ -8,6 +8,8 @@
         FactoryGirl.create(:offspring)
```
https://github.com/rails/rails/issues/18950#issuecomment-77924771
It is possible that controller tests are deprecated, something to follow and see if finally happens

alexvkcr replied:
Should we create a folder 'integration' for integration tests and deprecate controller test? or just keep and eye on it?

katafira commented:
@alexvkcr @delaluzparra

katafira commented:
Added lines to spec_helper.rb so that it prints the output of the tests in color on the terminal screen. Changed the grades tests for the model as @sergio-ocon  suggested. 
@alexvkcr @delaluzparra

katafira commented:
Added lines to spec_helper.rb so that it prints the output of the tests in color on the terminal screen. Changed the grades tests for the model as @sergio-ocon suggested.
@alexvkcr @delaluzparra

sergio-ocon commented on spec/spec_helper.rb:
```diff
@@ -31,6 +31,16 @@
+  config.color = true
+
+  # Use color not only in STDOUT but also in pagers and files
+  config.tty = true
```
You can configure this for your environment, you don't need to configure this for everyone else, and this should never be part of the original PR, as it is related to a completely different thing.
Now I can't merge your PR

sergio-ocon commented on spec/controllers/offsprings_controller_spec.rb:
```diff
@@ -23,4 +25,32 @@
+        sign_in user 
+        expect{
+          post :create, offspring: {first_name: "pepe", last_name: "kata", grade: :primary_second}
+        }.to change(user.offsprings, :count).by(0)
```
Improving a lot!

There is an error message that happens when you try to add an offspring other than first. 
You should test that too, if the user tries to add, he gets the message

alexvkcr replied:
The error message is a redirectioning to instructions page, we chaecked that the redirectioning is performed correctly on grades other than 'primary_first'

sergio-ocon commented on spec/controllers/offsprings_controller_spec.rb:
```diff
@@ -11,18 +11,14 @@
-        expect(assigns).to eq(nil) #it does not go through the controller
-        expect(response).to redirect_to(new_user_session_path)
-      end
+      # pending "redirects to authentication" 
```
Change it for something that makes sense. Open an issue if you have a suggestiion.
Move the code to where it belongs and call your PR "refactoring of"
Deleting the use case and making a comment there does not help anybody to know that there is an issue
A #pending will not be shown as pending in RSPEC
pending "redirects to authentication would"

sergio-ocon commented on spec/controllers/offsprings_controller_spec.rb:
```diff
@@ -11,18 +11,14 @@
-        get :index
-        expect(response).to have_http_status(:success)
-      end
+      # pending "shows index of offspring" 
```
You need to test the action in the controller
$bin/rake routes
And then see what routes are part of the offsprings controller (create, new, destroy)

sergio-ocon commented on spec/controllers/offsprings_controller_spec.rb:
```diff
@@ -37,20 +33,29 @@
           
       end
+      describe "and not in first grade" do
+        it "does not allow any other" do
```
????
context
it "fails when the grade is not 1st grade"...

sergio-ocon commented:
Rebase and republish

sergio-ocon commented:
Rebase and republish

alexvkcr commented:
Ready to merge

alexvkcr commented:
Ready to review, assigned to @katafira but anyone could review it

alexvkcr commented:
The PR became corrupted in the commit history after deleting config color changes in the old commit. Closing PR and openning a new one

alexvkcr commented:
It has been replaced by PR #13
